### PR TITLE
Add vmware_vm_vm_drs_rule module

### PR DESCRIPTION
**Pull request by machacekondra**

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - main
+permissions:
+  contents: read
+jobs:
+  ansible_test:
+    runs-on: ubuntu-latest
+    container: fedora:latest
+    services:
+      testvcenter:
+        image: vmware/vcsim
```
Comment (bardielle) [inline .github/workflows/integration-tests.yml:19]: I think it should be a separate vcsim env for each test (as a prepare_lab role )

File: tests/integration/targets/vmware_vm_vm_drs_rule/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+- name: Create a DRS Affinity rule for vms
```
Comment (bardielle) [inline tests/integration/targets/vmware_vm_vm_drs_rule/tasks/main.yml:1]: This test should use a role for preparing a mock env

File: tests/integration/integration_config.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+vcenter_hostname: "testvcenter"
```
Comment (bardielle) [inline tests/integration/integration_config.yml:1]: where do you set those attributes? for example if we need DC ?

File: tests/integration/targets/vmware_vm_vm_drs_rule/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    drs_rule_name: drs_rule_0001
+    cluster_name: "{{ cluster_name }}"
+    vms: "{{ vms }}"
+    enabled: true
+    affinity_rule: true
+    mandatory: true
+  register: drs_rule_result
+  check_mode: true
+
+- name: Should report changed=true
```
Comment (bardielle) [inline tests/integration/targets/vmware_vm_vm_drs_rule/tasks/main.yml:17]: I think you missed the update and delete

File: tests/integration/integration_config.yml (hunk 3 above)
Comment (machacekondra) [reply tests/integration/integration_config.yml:1]: Currently it must be added manually here.

**Pull request closed without merging**